js/service/shell/index.js

```diff
@@ -44,3 +44,20 @@ exports.setCommand = function(name, cb) {
   assert(typeutils.isFunction(cb));
   commands.set(name, cb);
 };
+
+exports.runCommand = function(name, args, cb) {
+  var capture = require('./captureio.js');
+
+  assert(typeutils.isString(name));
+
+  args = args || "";
+
+  commands.get(name)(args, capture.io, function() { capture.events.emit('end'); });
+
+  if (cb) {
+    capture.events.redrawPrompt = inputBox.done;
+    cb(capture.events);
+  }
```

Comment:
the err parameter was not my main point - but rather that I don't understand why giving the possibility to use a cb is needed at all - it seem that you can use the returned value and that that'd be enough.